migen/test/test_vcd.py

```diff
@@ -0,0 +1,44 @@
+import unittest
+import os
+import tempfile
+
+from migen.sim.vcd import VCDWriter
+
+
+class VcdWriter(unittest.TestCase):
+
+    def get_file_path(self, dir):
+        return os.path.join(dir, self.id() + ".vcd")
+
+    def check_expectation(self, filename):
+        self.vcd.close()
+        with open(filename, 'r') as f:
+            self.assertMultiLineEqual(self.expected_file, f.read())
+
+    def test_empty(self):
+
+        self.expected_file = (
+            "$dumpvars\n"
+            "$end\n"
+            "#0\n"
+        )
+
+        with tempfile.TemporaryDirectory() as dir:
+            filename = self.get_file_path(dir)
+            self.vcd = VCDWriter(filename)
+            self.check_expectation(filename)
+
+    def test_module_name(self):
+
+        self.expected_file = (
+            "$scope module name1 $end\n"
+            "$enddefinitions $end\n"
+            "$dumpvars\n"
+            "$end\n"
+            "#0\n"
+        )
+
+        with tempfile.TemporaryDirectory() as dir:
+            filename = self.get_file_path(dir)
+            self.vcd = VCDWriter(filename, module_name="name1")
+            self.check_expectation(filename)
```

Comment:
Why use `self.expected_file`?
You could just do  `self.check_expectation(filename, expected_file)` (I would also call it `expected_contents`)